### PR TITLE
Allow analyzer 1.0

### PR DESCRIPTION
**Pull request by robbecker-wf**
## Motivation
  
Allow analyzer 1.0

## Changes
  

- Updated some ranges to get closer to resolving analyzer 1.0
- Add CI checks to validate each supported major version of analyzer

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - Verify all new "validate_analyzer" runs pass
        - Verify "validate_analyzer" resolves to the corresponding analyzer major version in each run (see `pub get` output)
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: pubspec.yaml [hunk 1]
```diff
@@ -56,3 +53,9 @@ dependency_validator:
     - tools/**
   ignore:
     - meta
+
+dependency_overrides:
+  platform_detect:
+    path: ../platform_detect
+  built_redux:
+    path: ../built_redux
```
Comment (robbecker-wf) [inline pubspec.yaml:61]: 7.5.13 but with `build: '>=1.2.2 <3.0.0'`

Comment (robbecker-wf) [reply pubspec.yaml]: released as 7.5.14

File: tool/validate_analyzer_version.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Fail this script if any command fails
+set -e
+# Print commands as they run
```
Comment (robbecker-wf) [inline tool/validate_analyzer_version.sh:5]: thank you for the comments!

**Comment by robbecker-wf** [comment]
Just 1 suggestion but everything else looks good.

**Review by greglittlefield-wf — DISMISSED** [review]
+1 to @robbecker-wf's commits

File: pubspec.yaml [hunk 3]
```diff
@@ -7,14 +7,11 @@ environment:
 
 dependencies:
   collection: ^1.14.11
-  # Dart 2.7 needs 0.39.x to resolve,
-  # and Dart 2.12+ needs 0.42.x to resolve to build_web_compilers 2.12.0, etc.
-  # to enable proper opting out of null-safety.
-  analyzer: '>=0.39.0 <0.42.0'
+  analyzer: '>=0.40.0 <2.0.0'
   build: '>=1.0.0 <3.0.0'
   built_redux: ^7.4.2
```
Comment (aaronlademann-wf) [inline pubspec.yaml:12]: `built_redux < 8.0.0` does not allow analyzer 1.0. Should this be expanded?

```suggestion
  built_redux: ">=7.4.2 <9.0.0"
```

Comment (greglittlefield-wf) [reply pubspec.yaml:12]: Based on [this CI run](https://github.com/Workiva/over_react/runs/6374857931?check_suite_focus=true) it seems like it does:
Screen Shot 2022-05-10 at 10 52 37 AM


Though, I'm not opposed to raising the bound on built_redux so this package isn't a bottleneck. Let me take a look

File: test/over_react/component_declaration/redux_component_test.dart [hunk 4]
```diff
@@ -78,14 +78,14 @@ void main() {
       var renderedInstance = render(TestDefault()..store = store);
       TestDefaultComponent component = getDartComponent(renderedInstance);
 
-      store.actions.trigger1();
+      store.actions.trigger1(null);
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/redux_component_test.dart:81]: The argument to `ActionDispatcher` was originally optional but then was made required in built_redux 8.0.0

File: pubspec.yaml (hunk 3 above)
Comment (robbecker-wf) [reply pubspec.yaml]: yeah we've backpatched built_redux v7 to allow analyzer v1

**Review by aaronlademann-wf — APPROVED** [review]
+1

**Comment by robbecker-wf** [comment]
+10 verified that all validate_analyzer jobs passed and are using the correct analyzer versions each.

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-p

**Pull request merged**